### PR TITLE
Poet listing

### PR DESCRIPTION
**Pull request by HinataIzayoi**
Add poets listing at /poets with query support. Logged in User can follow and unfollow poets. 
Add User profile page with User poems.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export default class App extends React.Component {
     }
   }
 
-  genericApi1(event, arg) {
+  genericAPI(event, arg) {
```
Comment (Kailang) [inline src/App.js]: Revert name change.

File: src/App.js [hunk 2]
```diff
@@ -101,15 +101,54 @@ export default class App extends React.Component {
...
+          throw new Error(e);
+        });
+    if (poets) {
+      return poets;
+    }
+  }
+
+  async userFollowUser({token, followId}) {
+    await this.genericAPI('/v1/users/follow', {token, followId})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Remove .catch.

File: src/App.js [hunk 3]
```diff
@@ -101,15 +101,54 @@ export default class App extends React.Component {
...
+          console.log(e);
+          throw new Error(e);
+        });
+  }
+
+  async userUnfollowUser({token, unfollowId}) {
+    await this.genericAPI('/v1/users/unfollow', {token, unfollowId})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Same.

File: src/App.js [hunk 4]
```diff
@@ -101,15 +101,54 @@ export default class App extends React.Component {
...
+          console.log(e);
+          throw new Error(e);
+        });
+  }
+
+  async userPoem({token, targetUser}) {
+    const poems = await this.genericAPI('/v1/poets/poems', {token, targetUser})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Same.

File: src/components/Poem.js [hunk 5]
```diff
@@ -17,22 +17,23 @@ export default class Poem extends React.Component {
...
     return <div class="My(50px)">
       <div class="C(lightgrey) Fz(14dpx)">
-        <i class="fas fa-fire"></i> {hotness}°
+        <i class="fas fa-fire"></i> {viewCount}°
       </div>
       <div>
-        {visibility &&
-        <span class="Bgc(lightgray) D(ib) Px(4px) Py(0) Fz(10px) Bdrs(2px) Mend(10px)">{visibility}</span>}
+        {isOwner &&
+        <span class="Bgc(lightgray) D(ib) Px(4px) Py(0) Fz(10px) Bdrs(2px) Mend(10px)">{privacy}</span>}
```
Comment (Kailang) [inline src/components/Poem.js]: Add an indent for this line since it is under a {}

File: src/components/PoemInfo.js [hunk 6]
```diff
@@ -12,7 +12,9 @@ export default class PoemInfo extends React.Component {
           <div class="D(ib) Va(m) Fz(12px) C(gray)">
              Written by <br/><span class="Fz(16px) C(darkred) Td(u):h">{author}</span>
           </div>
+          {!isOwner&&
```
Comment (Kailang) [inline src/components/PoemInfo.js]: `{!isOwner &&`

File: src/components/PoemInfo.js [hunk 7]
```diff
@@ -12,7 +12,9 @@ export default class PoemInfo extends React.Component {
...
           <span class="Fl(end) Bgc(lightgray) Bgc(darkred):h C(white):h Px(8px) Py(2px) Mt(10px) Mend(8px) Fz(12px) Bdrs($bdrs-control)">Follow</span>
```
Comment (Kailang) [inline src/components/PoemInfo.js]: Add an indent.

File: src/components/PoetListingPage.js [hunk 8]
```diff
@@ -22,31 +17,74 @@ export default class PoetListingPage extends React.Component {
...
+        token: this.app.state.token,
+        filter: FILTER_ALL,
+        sort: this.state.sort,
+        order: this.state.order,
+        limit: DEFAULT_LIMIT,
+        skip: this.state.skip,
+        activeYearLImit: this.state.year,
+        search: this.state.q,
+      });
+      if (poets) {
```
Comment (Kailang) [inline src/components/PoetListingPage.js]: Move the if clause out of the try block.

File: src/components/PoetListingPage.js [hunk 9]
```diff
@@ -22,31 +17,74 @@ export default class PoetListingPage extends React.Component {
...
+        this.setState({poets: poets.payload});
+      }
+    } catch (err) {
+      console.log('error');
```
Comment (Kailang) [inline src/components/PoetListingPage.js]: An auto-retry would be ideal.
For now, console.log(err); would be fine.

File: src/utils.js [hunk 10]
```diff
@@ -127,3 +129,19 @@ export const BAD_REQUEST = 400;
 export const UNAUTHORIZED = 401;
 export const FORBIDDEN = 403;
 export const NOT_FOUND = 404;
+
+export function short(poemBody) {
```
Comment (Kailang) [inline src/utils.js]: Use getExcerpt()

File: src/utils.js [hunk 11]
```diff
@@ -2,6 +2,8 @@ import queryString from 'query-string';
...
 
+const POEM_SHORT_LENGTH = 100;
```
Comment (Kailang) [inline src/utils.js]: POEM_EXCERPT_LENGTH

File: src/App.js [hunk 12]
```diff
@@ -101,15 +101,54 @@ export default class App extends React.Component {
     }
   }
 
+  async userList({token, filter, sort, order, limit, skip, activeYearLimit, search}) {
+    const poets = await this.genericAPI('/v1/poets', {token, filter, sort, order, limit, skip, activeYearLimit, search})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Remove catch.

File: src/App.js (hunk 1 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 2 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 3 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 4 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/components/Poem.js (hunk 5 above)
Comment (HinataIzayoi) [reply src/components/Poem.js]: Done.

File: src/components/PoemInfo.js (hunk 7 above)
Comment (HinataIzayoi) [reply src/components/PoemInfo.js]: Done.

File: src/components/PoetListingPage.js (hunk 9 above)
Comment (HinataIzayoi) [reply src/components/PoetListingPage.js]: Done.

File: src/utils.js (hunk 10 above)
Comment (HinataIzayoi) [reply src/utils.js]: Done.

File: src/utils.js (hunk 11 above)
Comment (HinataIzayoi) [reply src/utils.js]: Done.

File: src/App.js (hunk 12 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

**Comment by HinataIzayoi** [comment]
Moved changes to hinata-user

**Pull request closed without merging**